ui/src/router/components/form/components/router_config/rule_card/RuleCard.js

```diff
@@ -150,56 +173,68 @@ export const RuleCard = ({
         }
         fullWidth>
         <Fragment>
-          {[...rule.routes, "_none_"].map((route, idx) => (
-            <Fragment key={`rule-routes-${idx}`}>
-              <EuiFlexGroup
-                className="euiFlexGroup--trafficRulesRow"
-                direction="row"
-                gutterSize="m"
-                alignItems="flexStart">
-                <EuiFlexItem grow={true} className="euiFlexItem--content">
-                  <EuiFormRow
-                    isInvalid={!!get(errors, `routes.${idx}`)}
-                    error={
-                      Array.isArray(get(errors, `routes.${idx}`))
-                        ? get(errors, `routes.${idx}`)
-                        : []
-                    }
-                    fullWidth>
-                    <EuiSuperSelect
-                      fullWidth
-                      hasDividers
-                      options={
-                        idx < rule.routes.length
-                          ? routesOptions(route)
-                          : [noneRoute, ...routesOptions(route)]
-                      }
-                      valueOfSelected={route}
-                      onChange={onChange(`routes.${idx}`)}
+        {[...rule.routes, "_none_"].map((route, idx) => {
+          // Hide last row when all routes are selected
+          if (idx < routes.length) {
```

Comment:
Can we instead control this by only adding the `__none__` above if `rule.routes.length < routes.length` ? I feel that is more intentional.